src/camera_capture/camera_capture/camera_capture.py

```diff
@@ -0,0 +1,206 @@
+    """this class captures immages from a CSI camera and publishes it as a livestream or snapshots to a topic"""
+
+    def __init__(
+        self,
+        publish_livestream_topic: str,
+        publish_snapshot_topic: str,
+        snapshot_trigger_topic: str,
+        livestream_state_topic: str,
+    ):
+        super().__init__(PACKAGE_NAME)
+
+        # initialize publisher & subscirbers
+        self.pub_cam_snapshot = self.create_publisher(Image, publish_snapshot_topic, 1)
+        self.pub_cam_livestream = self.create_publisher(
+            Image, publish_livestream_topic, 1
+        )
+        self.sub_cam_livestream_state = self.create_subscription(
+            Bool, livestream_state_topic, self.start_livestream_callback, 1
+        )
+        self.sub_cam_snapshot_trigger = self.create_subscription(
+            Bool, snapshot_trigger_topic, self.capture_snapshot_callback, 1
+        )
+
+        self.cap = cv2.VideoCapture(gstreamer_pipeline())
+        self.bridge = CvBridge()
+        self.image_location = f"{json_settings['image_location']}"
+        self.image_counter = 0
+
+    def start_livestream_callback(self, topic_msg: Bool):
+        """This method starts a livestream when a message is received on the topic livestream/state
+
+        Args:
+        topic_msg (Bool): True for starting livestream, False for stopping livestream
+        """
+        self.cap.set(
+            cv2.CAP_PROP_FRAME_HEIGHT, json_settings["capture_height_livestream"]
+        )
+        self.cap.set(
+            cv2.CAP_PROP_FRAME_WIDTH, json_settings["capture_width_livestream"]
+        )
+        self.get_logger().info(f"message received on topic livestream")
+        self.get_logger().debug(
+            f"Incomming message on topic livestream \nwith message: {topic_msg}"
+        )
+        timer_period = 0.03  # seconds TODO: make into settings 30hz
+        if self.cap.isOpened():
+            self.get_logger().info("camera is available")
+            if topic_msg.data:
+                self.get_logger().info("starting livestream")
+                timer = self.create_timer(timer_period, self.timer_callback)
+            elif not topic_msg.data:
+                self.get_logger().info("stopping livestream")
+                try:
+                    timer.cancel()
+                    timer.destroy()
+                except UnboundLocalError:
+                    self.get_logger().error("timer was is not defined)")
+            else:
+                self.get_logger().debug(
+                    "can't stop livestream because livestream is already stopped"
+                )
+        elif not self.cap.isOpened():
+            self.get_logger().info("camera not available")
+
+    def timer_callback(self):
+        """callback function to read image data from camera and publish it to the topic of the livestream"""
+        ret, frame = self.cap.read()
+        msg_image = self.bridge.cv2_to_imgmsg(np.array(frame), "bgr8")
```

Comment:
Is the `np.array()` needed? I think it is superfluous, if I recall correctly we implemented it to cast the frame to a certain `dtype` which turned out to be unnecessary.